client/src/index.js

```diff
@@ -2,27 +2,32 @@ import React from 'react'
 import ReactDOM from 'react-dom'
 import './index.css'
 import App from './App'
-import registerServiceWorker from './registerServiceWorker'
 
 import Loadable from 'react-loadable'
 import { Provider as ReduxProvider } from 'react-redux'
-import configureStore from './store/configureStore'
+import { configureStore } from './store'
 
-const store = configureStore( window.REDUX_STATE || {} )
-
-const AppBundle = (
-  <ReduxProvider store={store}>
-    <App />
-  </ReduxProvider>
-)
-
-window.onload = () => {
-  Loadable.preloadReady().then(() => {
-    ReactDOM.hydrate(
-      AppBundle,
-      document.getElementById('root')
-    )
-  })
+if (typeof window !== 'undefined') {
+  const store = configureStore(window.REDUX_STATE || {})
+  
+  const AppBundle = (
+    <ReduxProvider store={store}>
+      <App />
+    </ReduxProvider>
+  )
+  
+  window.onload = () => {
+    Loadable.preloadReady().then(() => {
+      ReactDOM.hydrate(
+        AppBundle,
+        document.getElementById('root')
+      )
+    })
+  }
+  
+  import('./registerServiceWorker')
+    .then(({default: register}) => register())
 }
 
-registerServiceWorker()
+export {App}
+export * from './store'
```

Comment:
In order for the web application to be both:
1. runnable in browser
2. imported from server code

I had to wrap all browser related code (including service worker which cannot load on server side) and to expose everything server needs (App class + store functions)